Client/Frontend/Browser/LoginsHelper.swift

```diff
@@ -248,20 +234,38 @@ class LoginsHelper: TabContentScript {
         tab?.addSnackbar(snackBar!)
     }
 
-    fileprivate func requestLogins(_ login: LoginRecord, requestId: String) {
-        profile.logins.getLoginsForProtectionSpace(login.protectionSpace).uponQueue(.main) { res in
-            var jsonObj = [String: Any]()
-            if let cursor = res.successValue {
-                log.debug("Found \(cursor.count) logins.")
-                jsonObj["requestId"] = requestId
-                jsonObj["name"] = "RemoteLogins:loginsFound"
-                jsonObj["logins"] = cursor.map { $0!.toJSONDict() }
+    fileprivate func requestLogins(_ request: [String : Any], url: URL) {
+        guard let requestId = request["requestId"] as? String,
+            // Even though we don't currently use these two fields,
+            // verify that they were received as additional confirmation
+            // that this is a valid request from LoginsHelper.js.
+            let _ = request["formOrigin"] as? String,
+            let _ = request["actionOrigin"] as? String,
+
+            // We pass in the webview's URL and derive the origin here
+            // to workaround Bug 1194567.
+            let origin = getOrigin(url.absoluteString) else {
+            return
+        }
+
+        let protectionSpace = URLProtectionSpace.fromOrigin(origin)
+
+        profile.logins.getLoginsForProtectionSpace(protectionSpace).uponQueue(.main) { res in
+            guard let cursor = res.successValue else {
+                return
             }
 
-            let json = JSON(jsonObj)
-            let src = "window.__firefox__.logins.inject(\(json.stringify()!))"
-            self.tab?.webView?.evaluateJavaScript(src, completionHandler: { (obj, err) -> Void in
-            })
+            log.debug("Found \(cursor.count) logins.")
+
+            let dict: [String : Any] = [
+                "requestId": requestId,
+                "name": "RemoteLogins:loginsFound",
+                "logins": cursor.compactMap({ $0?.toJSONDict() })
+            ]
+
+            let json = JSON(dict)
+            let injectJavaScript = "window.__firefox__.logins.inject(\(json.stringify()!))"
```

Comment:
`if let` here to avoid the `!` ?

Reply:
I was going to, but the old code did this too. It really should never fail and if it does, we'll at least be able to see it in crash reports.